codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -272,24 +254,17 @@ private int minUsableChannelBytes() {
         // an "adequate" amount of connection window before allocation is attempted. This is not foolproof as if the
         // number of streams is >= this minimal number then we may still have the issue, but the idea is to narrow the
         // circumstances in which this can happen without rewriting the allocation algorithm.
-        return Math.max(ctx.channel().config().getWriteBufferLowWaterMark(), MIN_WRITABLE_CHUNK);
+        return max(ctx.channel().config().getWriteBufferLowWaterMark(), MIN_WRITABLE_CHUNK);
     }
 
     private int maxUsableChannelBytes() {
-        if (ctx == null) {
-            return 0;
-        }
-
         // If the channel isWritable, allow at least minUseableChannelBytes.
-        int channelWritableBytes = (int) Math.min(Integer.MAX_VALUE, ctx.channel().bytesBeforeUnwritable());
-        int useableBytes = channelWritableBytes > 0 ? max(channelWritableBytes, minUsableChannelBytes()) : 0;
-
-        // Clip the usable bytes by the connection window.
-        return min(connectionState().windowSize(), useableBytes);
```

Comment:
Yes. It was a duplicate operation because `writableBytes` is always called with `requestedBytes` == `connection window` and so the clipping is already done.